packages/synthetics-sdk-broken-links/src/broken_links.ts

```diff
@@ -56,59 +68,124 @@ export enum StatusClass {
-    await origin_page.waitForSelector(options.wait_for_selector);
-  }
+    // check all links
+    followed_links.push(...(await checkLinks(browser, linksToFollow, options)));
 
-  // TODO
-  // scrape origin_url for all links
-  try {
-    // eslint-disable-next-line  @typescript-eslint/no-unused-vars
-    const retrieved_links: LinkIntermediate[] = await retrieveLinksFromPage(
-      origin_page,
-      options.query_selector_all,
-      options.get_attributes
+    // returned a SyntheticResult with `options`, `followed_links` &
+    // runtimeMetadata
+    return createSyntheticResult(
+      startTime,
+      runtime_metadata,
+      options,
+      followed_links
     );
-
-    // TODO shuffle links if link_order is `RANDOM`
-    // TODO always truncate to lint_limit
   } catch (err) {
-    if (err instanceof Error) process.stderr.write(err.message);
-    // TODO throw generic error with `failure to scrape links`
+    const errorMessage =
+      err instanceof Error
+        ? err.message
+        : `An error occurred while starting or running the broken link checker on ${inputOptions.origin_url}. Please reference server logs for further information.`;
+    return getGenericSyntheticResult(startTime, errorMessage);
+  } finally {
+    if (browser! !== undefined) await closeBrowser(browser!);
   }
+}
 
-  // TODO
-  // create new page to be used for all scraped links
-  // navigate to each link - LOOP:
-  //          each call to `checkLink(...)` will return a `SyntheticLinkResult`
-  //          Object added to an array of `followed_links`
-  const followed_links: BrokenLinksResultV1_SyntheticLinkResult[] = [];
-
-  // returned a SyntheticResult with `options`, `followed_links` &
-  // runtimeMetadata
-  return createSyntheticResult(
-    start_time,
-    runtime_metadata,
+/**
+ * Checks the origin link and returns the result.
+ *
+ * @param originPage - The Puppeteer page object representing the origin page.
+ * @param options - The broken link checker options.
+ * @returns The result of checking the origin link.
+ */
+async function checkOriginLink(
+  originPage: Page,
+  options: BrokenLinksResultV1_BrokenLinkCheckerOptions
+): Promise<BrokenLinksResultV1_SyntheticLinkResult> {
+  // check origin_link
+  const originLinkResult = await checkLink(
+    originPage,
+    { target_url: options.origin_url, anchor_text: '', html_element: '' },
     options,
-    followed_links
+    true
   );
+  return originLinkResult;
+}
+
+/**
+ * Scrapes links from the origin page and returns them.
+ * If applicable:
+ *     - wait for `options.wait_for_selector` element before scraping.
+ *     - shuffle and truncate based on `options`
+ *
+ * @param originPage - The Puppeteer page object representing the origin page.
+ * @param options - The broken link checker options.
+ * @returns An array of scraped links in accordance with link_limit and link_order.
+ */
+async function scrapeLinks(
```

Comment:
should this be in link utils? (by nature of being called, scrape links :))

Reply:
I was trying to keep all the helpers that are there purely to make `runBrokenLinks` more readable in broken_links.ts

What do you think? These methods are not independently tested but tested through the `runBrokenLinks` tests